### PR TITLE
460/bit song/nft tx

### LINKED ISSUE #460: Bitsong: NFT TX
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by yayay927**
closes #460 

## Checklist
- [x] Ran Linting
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: public/locales/en/message_contents.json [hunk 1]
```diff
@@ -49,5 +49,10 @@
   "txVersionContent": "<0>{{signer}}</0> submitted IBC version",
   "txTransferContent": "<0>{{sender}}</0> sent <2>{{token}}</2> via IBC transfer from source channel <2>{{sourceChannel}}</2> to <1>{{receiver}}</1>",
   "txTimeoutContent": "<0>{{signer}}</0> IBC timeout",
-  "txTimeoutOnCloseContent": "<0>{{signer}}</0> IBC timeout on close"
+  "txTimeoutOnCloseContent": "<0>{{signer}}</0> IBC timeout on close",
+  "txIssueDenomContent": "<0>{{creators}}</0> issued NFT <1>{{name}}</1> for <2>{{split_shares}}</2>",
```
Comment (ryuash) [inline public/locales/en/message_contents.json:53]:
```
// MsgIssueDenom defines an SDK message for creating a new denom.
type MsgIssueDenom struct {
	Id           string                                   `protobuf:"bytes,1,opt,name=id,proto3" json:"id,omitempty"`
	Name         string                                   `protobuf:"bytes,2,opt,name=name,proto3" json:"name,omitempty"`
	Creators     []string                                 `protobuf:"bytes,3,rep,name=creators,proto3" json:"creators,omitempty"`
	SplitShares  []github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,4,rep,name=split_shares,json=splitShares,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"split_shares" yaml:"split_shares"`
	RoyaltyShare github_com_cosmos_cosmos_sdk_types.Dec   `protobuf:"bytes,5,opt,name=royalty_share,json=royaltyShare,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"royalty_share" yaml:"royalty_share"`
	Sender       string                                   `protobuf:"bytes,6,opt,name=sender,proto3" json:"sender,omitempty"`
}
```

I don't think this has anything to do with NFT. It's just creating a new denom

File: src/models/msg/nft/msg_issue_denom.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+class MsgIssueDenom {
+  public category: Categories;
+  public type: string;
+  public json: any;
+  public id: string;
+  public name: string;
+  public creators: string[];
+  public splitShares: MsgCoin;
+  public royaltyShare: MsgCoin;
```
Comment (ryuash) [inline src/models/msg/nft/msg_issue_denom.ts:12]: I don't think the type is `MsgCoin`

```
// NOTE: never use new(Dec) or else we will panic unmarshalling into the
// nil embedded big.Int
type Dec struct {
	i *big.Int
}
```

---

We don't need to log the whole model. Just whatever the tx message needs

**Review by ryuash — CHANGES_REQUESTED** [review]

File: i18n.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ module.exports = {
   locales: ['en'],
   defaultLocale: 'en',
   pages: {
-    '*': ['common'],
+    '*': ['common', 'message_labels', 'message_contents', 'transactions'],
```
Comment (ryuash) [inline i18n.js:6]:
```
'*': ['common']
```

File: src/components/msg/nft/transfer_nft/index.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          ),
+          <b />,
+          (
+            <Name
+              address={message.recipient}
+              name={recipientMoniker}
+            />
+          ),
+          <b />,
+        ]}
```
Comment (ryuash) [inline src/components/msg/nft/transfer_nft/index.tsx:39]:
```
"txTransferNFTContent": "<0>{{sender}}</0> sent NFT <1>{{id}}</1> to <2>{{recipient}}</2>",
```

there only 3 variables here why is there 4 in this component

File: src/components/msg/utils.tsx [hunk 5]
```diff
@@ -372,6 +372,17 @@ const getDataByType = (type: string) => {
...
+    // ========================
+    // nft
+    // ========================
+    '/bitsong.nft.v1beta1.MsgIssueDenom': {
+      model: MODELS.MsgIssueDenom,
+      content: COMPONENTS.IssueDenom,
+      tagTheme: 'five',
+      tagDisplay: 'txIssueDenomLabel',
+    },
+
```
Comment (ryuash) [inline src/components/msg/utils.tsx:385]: Where are the other ones?

Comment (yayay927) [reply src/components/msg/utils.tsx:385]: Add now.

**Pull request merged**